Package.swift

```diff
@@ -13,35 +15,193 @@ let package = Package(
             name: "StreamChat",
             targets: ["StreamChat"]),
         .library(
-            name: "StreamChatCore",
-            targets: ["StreamChatCore"]),
-        .library(
-            name: "StreamChatClient",
-            targets: ["StreamChatClient"]),
+            name: "StreamChatUI",
+            targets: ["StreamChatUI"]
+        )
     ],
     dependencies: [
-        // UI
-        .package(url: "https://github.com/kean/Nuke.git", from: "8.4.0"),
-        .package(url: "https://github.com/SnapKit/SnapKit.git", from: "5.0.0"),
-        .package(url: "https://github.com/kirualex/SwiftyGif.git", from: "5.2.0"),
-        .package(url: "https://github.com/RxSwiftCommunity/RxGesture.git", from: "3.0.0"),
-        // Core
-        .package(url: "https://github.com/ReactiveX/RxSwift.git", from: "5.1.0"),
-        // Client
+        // StreamChat
         .package(url: "https://github.com/daltoniam/Starscream.git", from: "4.0.0"),
+        
+        // StreamChatUI
+        .package(url: "https://github.com/kean/Nuke.git", from: "9.0.0"),
+        .package(url: "https://github.com/kirualex/SwiftyGif.git", from: "5.3.0")
     ],
     targets: [
         .target(
             name: "StreamChat",
-            dependencies: ["StreamChatCore", "Nuke", "SnapKit", "SwiftyGif", "RxGesture"],
-            path: "Sources/UI"),
-        .target(
-            name: "StreamChatCore",
-            dependencies: ["StreamChatClient", "RxSwift", .product(name: "RxCocoa", package: "RxSwift")],
-            path: "Sources/Core"),
-        .target(
-            name: "StreamChatClient",
             dependencies: ["Starscream"],
-            path: "Sources/Client")
+            exclude: ["README.md", "Info.plist"] + streamChatSourcesExcluded,
+            resources: [.copy("Database/StreamChatModel.xcdatamodeld")]
+        ),
+        .target(
+            name: "StreamChatUI",
+            dependencies: ["StreamChat", "Nuke", "SwiftyGif"],
+            exclude: ["README.md", "Info.plist"] + streamChatUISourcesExcluded,
+            resources: [.process("Sources/StreamChatUI/Resources")]
```

Comment:
Does it work when you don't put `path: "Sources/StreamChatUI"`? I didn't know SPM was that smart

Reply:
Yes, the default path for sources is `Sources/<target name>`. fun fact: It used to be the only way the first couple of years of SPM. You couldn't even specify your custom folder name.